modules/tracking/src/roiSelector.cpp

```diff
@@ -163,7 +163,7 @@ namespace cv {
     // while key is not ESC (27)
     while(key!=27){
       temp=select(windowName, img, true, fromCenter);
-      if(temp.width>0 && temp.height>0)
+      if(key!=27 && temp.width>0 && temp.height>0)
```

Comment:
Ow well I didnt look close enough to the fact that there was indeed a function call that could change the key value. In that case it would indeed be better to switch the checks to after the function call like you suggested. That solves the problem at all.